plugin/storage/integration/integration_test.go

```diff
@@ -0,0 +1,276 @@
+		actual, err := s.reader.GetServices()
+		require.NoError(t, err)
+		if found = assert.ObjectsAreEqualValues(expected, actual); found {
+			break
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetOperations(t *testing.T) {
+	t.Log("Testing GetOperations ...")
+
+	expected := []string{"example-operation-1", "example-operation-3", "example-operation-4"}
+	s.getBasicFixture(t)
+
+	s.refresh()
+
+	var found bool
+	var actual []string
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err := s.reader.GetOperations("example-service-1")
+		require.NoError(t, err)
+		if found = assert.ObjectsAreEqualValues(expected, actual); found {
+			break
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetTrace(t *testing.T) {
+	t.Log("Testing GetTrace ...")
+
+	expected := s.getBasicFixture(t)
+	expectedTraceID := model.TraceID{Low: 17}
+
+	s.refresh()
+
+	var found bool
+	var actual *model.Trace
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err := s.reader.GetTrace(expectedTraceID)
+		if err == nil {
+			if found = len(actual.Spans) == len(expected.Spans); found {
+				break
+			}
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	assert.True(t, found)
+	CompareTraces(t, expected, actual)
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestFindTraces(t *testing.T) {
+	t.Log("Testing FindTraces ...")
+	t.Log("\t(Note: all cases include ServiceName + StartTime range)")
+	queries, err := getQueries()
+	require.NoError(t, err)
+	for _, query := range queries {
+		t.Logf("\t\t* Query case: + %s", query.Caption)
+		s.integrationTestFindTracesByQuery(t, query.Query, query.ExpectedFixture)
+	}
+}
+
+func (s *StorageIntegration) integrationTestFindTracesByQuery(t *testing.T, query *spanstore.TraceQueryParameters, expectedFixture string) {
+	expectedTrace, err := getTraceFixture(expectedFixture)
+	require.NoError(t, err)
+	require.NoError(t, s.writeTrace(expectedTrace))
+	expected := []*model.Trace{expectedTrace}
+	require.NoError(t, s.refresh())
+
+	var found bool
+	var actual []*model.Trace
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err = s.reader.FindTraces(query)
+		if err == nil {
+			if len(actual) == query.NumTraces {
+				found = true
+				break
+			}
+			found = tracesMatch(actual, expected)
```

Comment:
why can't we do:

```
if tracesMatch(actual, expected) {
     break
}
```

and then CompareSliceOfTraces below after the for loop once? Given CompareSliceOfTraces uses `require` will stop execution anyway for bad traces, just do it once